dao/mongo.go

```diff
@@ -672,39 +673,10 @@ func (m *MongoService) GetStatementOfAffairsResource(transactionID string) (mode
 
 // DeleteStatementOfAffairsResource deletes the statement of affairs filed for an insolvency case
 func (m *MongoService) DeleteStatementOfAffairsResource(transactionID string) (int, error) {
-	collection := m.db.Collection(m.CollectionName)
-
-	// Choose specific transaction for insolvency case with attachment to be removed
-	filter := bson.M{"transaction_id": transactionID}
-
-	// Check if insolvency case exists for specified transactionID
-	storedInsolvency := collection.FindOne(context.Background(), filter)
-	err := storedInsolvency.Err()
-	if err != nil {
-		if err == mongo.ErrNoDocuments {
-			log.Error(err)
-			return http.StatusNotFound, fmt.Errorf(constants.MsgCaseForTransactionNotFound, transactionID)
-		}
-		log.Error(err)
-		return http.StatusInternalServerError, fmt.Errorf(constants.MsgHandleReqTransactionId, transactionID)
-	}
 
-	query := bson.M{"data.statement-of-affairs": ""}
+	httpStatus, err := m.DeleteResourceDBChecks(transactionID, "statement-of-affairs")
```

Comment:
You could use the attachmentTyoe constant

Reply:
Complicates it because you would have to do further mapping as we are attempting to delete the collection which can differ from the attachment type, e.g. collection = statement-of-affairs & attachmentType = statement-of-affairs-director